enhancements/network/ovn-secondary-networks.md

```diff
@@ -0,0 +1,389 @@
+---
+title: ovn-secondary-networks
+authors:
+  - "@maiqueb"
+reviewers:
+  - "@dcbw"
+  - "@dougbtv"
+  - "@squeed"
+  - "@trozet"
+  - "@abhat"
+  - "@alexanderConstantinescu"
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-06-15
+last-updated: yyyy-mm-dd
+status: provisional
+---
+
+# OVN secondary networks
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+This enhacement proposal studies and describes how to support OVN as the
+networking provider for a Multus secondary interface.
+
+## Motivation
+The networking plugins that can be used as secondary interfaces are currently
+quite simple, and are unable to provide advanced networking requirements.
+
+Furthermore, virtualization users are used to being able to define isolated
+tenant networks that form a single L2 domain. OVN excels at this, by creating
+simple L2 overlays without the need for a network administrator.
```

Comment:
I think we might be missing some items in this design proposal around the virtualization (CNV) use case, specifically how to support Live Migration of KubeVirt VMs with static IP addresses.

For example, if we assign a static IP address to a KubeVIrt VMI's pod, during live migration there will temporarily be a secondary target pod on another node that acts as the destination for the live migration. So, for a brief period of time, we'll need two Pods for the VMI to be running at the same time.

How will the scenario of two pods temporarily needing the same IP work in this design?

We need this static IP assignment to work with Live Migration, otherwise the benefit here is minimal compared to using the standard default pod network today.

Reply:
@davidvossel what is an acceptable downtime for the service ? 

As Kubernetes is concerned, it does not care the same IP is allocated in multiple pods at once. 

As for OVN is concerned, this is doable - migrations (on oVirt, and Openstack) are done by updating the logical port `requested-chassis` to the value of the destination chassis. (chassis = OVN node, so to speak ..)

There's an [on-going effort](https://patchwork.ozlabs.org/project/ovn/list/?series=298609&state=*) to improve the downtime of the migration, but, migrating VMs should be possible out of the box. 

I can add a section indicating we would need to update an existing port rather than create a new one, explaining this in more detail.

Reply:
>@davidvossel what is an acceptable downtime for the service ? 

I'd like to target network interruption of less than 1 second. Is that a practical expectation?

Reply:
>@davidvossel what is an acceptable downtime for the service ? 

I'd like to target network interruption of less than 1 second. Is that a practical expectation?

Reply:
> > @davidvossel what is an acceptable downtime for the service ?
> 
> I'd like to target network interruption of less than 1 second. Is that a practical expectation?

I think it is reasonable, once the aforementioned patch-set is merged.

Reply:
presentation about this feature: 
 - https://youtu.be/ijZTMXAg-eI
 - https://www.openvswitch.org/support/ovscon2022/slides/Live-migration-with-OVN.pdf